### ISSUE TITLE
Infer metadata from tokens found in local transactions

### ISSUE THREAD

**Issue by mhluongo** [tallycash/extension#1021]

**Comment by VladUXUI** [comment]
Going to pick the second bullet up tomorrow. I have a couple of solutions for the 4 in mind but i don't want to rush it.

Can you say a bit more about "Add a way to see every asset on Etherscan"?

**Comment by mhluongo** [comment]
Yep! In particular, for an ERC-20 I want to see my balance + see what the contract is to make sure it's the token I want.

**Comment by VladUXUI** [comment]
Hopefully i understood this correctly.
Basically you will see a warning that something is up with this asset. On the click of the asset we will let you know the name/contract address and what TX this is coming from. This way you have several ways to verify.

This is for the first implementation, with two entry points to opening the slide-up. From asset page and asset details page.

![image](https://user-images.githubusercontent.com/80846886/157210147-ee08fe73-dc76-41d5-aa8e-fcf48428d00b.png)

**Comment by VladUXUI** [comment]
As a second implementation for _Eventually add a way for users to say "trust this token" or "ignore token" on the asset page for all auto-inferred tokens_ we would only need to add two buttons to the slide-up. (This version requires a bit of cleaning, but sharing for visibility)

![image](https://user-images.githubusercontent.com/80846886/157210421-595dcb42-ee4c-4b1b-81ed-67b65986b014.png)

**Comment by mhluongo** [comment]
Love love love this @VladUXUI.

**Comment by mhluongo** [comment]
I think adding a link out to EtherScan for every asset page is the only thing this is missing design-wise.

**Comment by mr-michael** [comment]
+ Toggle: "Display untrusted tokens" (default OFF)

**Comment by puppycodes** [comment]
To breakdown into newer tasks

**Comment by Shadowfiend** [comment]
Shipped in #3424 .
